File: Php/app/User/UserFactory.php
namespace app\User{

	use Domain\Classes\User;

	interface UserFactory{

		/**
		 * @param $id
		 * @param $userName
		 * @param $displayName
		 * @param $realName
		 * @param $email
		 * @param $globalRole
		 * @return User
		 */
		public function createUser($id, $userName, $displayName, $realName, $email, $globalRole);

	}

}

File: Infrastructure.Data/Factories/UserFactory.cs
using Core.Classes;
using Infrastructure.Data.Classes;
using Infrastructure.Data.Factories.Interfaces;

namespace Infrastructure.Data.Factories{

    public class UserFactory : IUserFactory{

        public User Create(RawUser rawUser)
        {
            return new User
            (
                rawUser.Id,
                rawUser.UserName,
                rawUser.DisplayName,
                rawUser.RealName,
                rawUser.Email,
                (Role)rawUser.GlobalRole,
                rawUser.Token,
                rawUser.EncryptedPassword,
                rawUser.Salt
            );
        }
    }

}